templates/mod/passenger.conf.erb

```diff
@@ -1,61 +1,262 @@
 # The Passenger Apache module configuration file is being
 # managed by Puppet and changes will be overwritten.
 <IfModule mod_passenger.c>
-  <%- if @passenger_root -%>
-  PassengerRoot "<%= @passenger_root %>"
+  <%- if @passenger_allow_encoded_slashes -%>
+  PassengerAllowEncodedSlashes <%= @passenger_allow_encoded_slashes %>
   <%- end -%>
-  <%- if @passenger_ruby -%>
-  PassengerRuby "<%= @passenger_ruby %>"
+  <%- if @passenger_app_env -%>
+  PassengerAppEnv <%= @passenger_app_env %>
+  <%- end -%>
+  <%- if @passenger_app_group_name -%>
+  PassengerAppGroupName <%= @passenger_app_group_name %>
+  <%- end -%>
+  <%- if @passenger_app_root -%>
+  PassengerAppRoot "<%= @passenger_app_root %>"
+  <%- end -%>
+  <%- if @passenger_app_type -%>
+  PassengerAppType <%= @passenger_app_type %>
+  <%- end -%>
+  <%- if @passenger_base_uri -%>
+  PassengerBaseURI <%= @passenger_base_uri %>
+  <%- end -%>
+  <%- if @passenger_buffer_response -%>
+  PassengerBufferResponse <%= @passenger_buffer_response %>
+  <%- end -%>
+  <%- if @passenger_buffer_upload -%>
+  PassengerBufferUpload <%= @passenger_buffer_upload %>
+  <%- end -%>
+  <%- if @passenger_concurrency_model -%>
+  PassengerConcurrencyModel <%= @passenger_concurrency_model %>
+  <%- end -%>
+  <%- if @passenger_data_buffer_dir -%>
+  PassengerDataBufferDir "<%= @passenger_data_buffer_dir %>"
+  <%- end -%>
+  <%- if @passenger_debug_log_file -%>
+  PassengerDebugLogFile <%= @passenger_debug_log_file %>
+  <%- end -%>
+  <%- if @passenger_debugger -%>
+  PassengerDebugger <%= @passenger_debugger %>
+  <%- end -%>
+  <%- if @passenger_default_group -%>
+  PassengerDefaultGroup <%= @passenger_default_group %>
   <%- end -%>
   <%- if @passenger_default_ruby -%>
   PassengerDefaultRuby "<%= @passenger_default_ruby %>"
   <%- end -%>
+  <%- if @passenger_default_user -%>
+  PassengerDefaultUser <%= @passenger_default_user %>
+  <%- end -%>
+  <%- if @passenger_disable_security_update_check -%>
+  PassengerDisableSecurityUpdateCheck <%= @passenger_disable_security_update_check %>
+  <%- end -%>
+  <%- if @passenger_enabled -%>
+  PassengerEnabled <%= @passenger_enabled %>
+  <%- end -%>
+  <%- if @passenger_error_override -%>
+  PassengerErrorOverride <%= @passenger_error_override %>
+  <%- end -%>
+  <%- if @passenger_file_descriptor_log_file -%>
+  PassengerFileDescriptorLogFile "<%= @passenger_file_descriptor_log_file %>"
```

Comment:
Some of the values may have white space. The passenger docs is not quote explicit on which ones can and can't.  The passenger docs specify in the syntax the kind of value and from there I wrote a conversion chart. Copy-paste from my script below:
```
type_conversion_chart = {
    'dir-path' => 'FullPath',
    'file-path' => 'FullPath',
    'on|off' => 'OnOff',
    'uri' => 'URI',
    'path-to-ruby-interpreter' => 'FullPath',
    'path-to-python-interpreter' => 'FullPath',
    'path-to-node-js' => 'FullPath',
    'path-to-json-settings-file' => 'FullPath',
    'name' => 'String',
    'path' => 'FullPath',
    'relative-path' => 'RelPath',
    'absolute-or-relative-path' => 'Path',
    'smart|direct' => %w(smart direct),
    'username' => 'String',
    'groupname' => 'String',
    'scheme://user:password@proxy_host:proxy_port' => 'URI',
    'integer' => 'Integer',
    'seconds' => 'Integer',
    'number' => 'Integer',
    'process|thread' => %w(process thread),
    'megabytes' => 'Integer',
    'url' => 'URI',
    'bytes' => 'Integer',
    'filename' => 'FullPath',
    'path-to-socket' => 'FullPath',
    'your_application_key' => 'String',
    '"string"' => 'QuotedString',
    'gateway_hostname' => 'URI',
    'gateway_port' => 'Integer',
    'gateway_pinned_certificate_file' => 'FullPath'
}


```

Here is the bit where at I decided what was what.
quoted_types = ['QuotedString', 'FullPath', 'Path', 'RelPath']
non_quoted_types = ['OnOff','Integer','String','URI']

Things I mapped as non_quoted_types are the only settings in which the passenger documentation made evident the value would not contain white space, everything else may contain white space.

The apache documetation states
> Arguments to directives are separated by whitespace. If an argument contains spaces, you must enclose that argument in quotes.

I also left in whatever quoting existed before my changes.  I didn't want to change anyone's module configuration who was using existing options with or without quotes.